### PR TITLE
Add MockBiomapper tests

### PR DESCRIPTION
**Pull request by tonycdot**

### CODE REVIEW

File: tests/hardhat/utils/FixRevertError.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  return new ContractFunctionRevertedError({
+    abi,
+    data: typeof data === "object" ? data.data : data,
+    functionName,
+    message: shortMessage ?? message,
+  });
+};
+
+export const fixRevertErrorIfNeeded = (err: unknown) => {
+  if (err instanceof ContractFunctionRevertedError) return err;
```
Comment (MOZGIII) [inline tests/hardhat/utils/FixRevertError.ts:25]: When does this happen?

**Review by MOZGIII — APPROVED** [review]

File: tests/hardhat/utils/FixRevertError.ts (hunk 1 above)
Comment (tonycdot) [reply tests/hardhat/utils/FixRevertError.ts:25]: Such behaviour is stated in the docs. It is done this way for more flexibility on potential `viem` updates.
Alternatively, we can only work with current situation and remove this function.

Comment (MOZGIII) [reply tests/hardhat/utils/FixRevertError.ts:25]: > stated in the docs
Link?

Comment (MOZGIII) [reply tests/hardhat/utils/FixRevertError.ts:25]: > on potential `viem` updates
Yeah, we'd want to disregard those that and fix our case. Here we aim to implement the most deterministic path possible as this is for the tests only.

Comment (tonycdot) [reply tests/hardhat/utils/FixRevertError.ts]: > Link?

I meant `viem` [docs](https://viem.sh/docs/contract/simulateContract#handling-custom-errors).

And I removed this code.

File: tests/hardhat/utils/FixRevertError.ts
[the remark this answers is not in the document]
Comment (tonycdot) [reply tests/hardhat/utils/FixRevertError.ts:1]: Done

File: tests/hardhat/utils/FixRevertError.ts (hunk 1 above)
Comment (MOZGIII) [reply tests/hardhat/utils/FixRevertError.ts]: I get it if you'd do it like this:

```ts
  if (err instanceof BaseError) {
    const revertError = err.walk(err => err instanceof ContractFunctionRevertedError)
    if (revertError instanceof ContractFunctionRevertedError) {
       return revertError;
    }
  }
  // ...
```

**Review by MOZGIII — APPROVED** [review]

**Review by sesh92 — APPROVED** [review]

**Pull request merged**